### ISSUE TITLE
State information for One-time connection establishment by using the out-of-band protocol is different between an agent sender and an agent receiver

### ISSUE THREAD

**Issue by yunxi-zhang** [hyperledger/aries-cloudagent-python#1157]
I have 2 ACA-Py agents, each runs in its own VM with multi-tenancy feature enabled. I've created 2 sub wallets in each agent. When trying to set up the agent-level connection between agent1's sub wallet 1 (as a sender) to  in agent2's sub wallet 1 (as a receiver) and sub wallet 2 (also as a second receiver), I created a connection invite in agent1 and consumed by  agent2's sub wallet 1, which works successfully. When the same connection invite is consumed by agent2's sub wallet 2, it returns error in the agent1's log as shown below.

```
2021-05-06 12:10:33,132 aries_cloudagent.connections.base_manager WARNING No corresponding DID found for sender verkey: G1jLBJbAmusmG3DGQ5k167YQ34ko28Kyd1ZchAbgK7F5
2021-05-06 12:10:33,198 aries_cloudagent.connections.base_manager WARNING No corresponding DID found for recipient verkey: EK9krLvVxnb45y6qBTYHrLj2vpGJaruwJFbnG39PqURB
2021-05-06 12:10:33,228 aries_cloudagent.messaging.base_handler ERROR Error receiving connection request
Traceback (most recent call last):
  File "/home/indy/aries_cloudagent/protocols/connections/v1_0/manager.py", line 479, in receive_request
    their_role=ConnRecord.Role.REQUESTER.rfc160,
  File "/home/indy/aries_cloudagent/connections/models/conn_record.py", line 290, in retrieve_by_invitation_key
    return await cls.retrieve_by_tag_filter(session, tag_filter, post_filter)
  File "/home/indy/aries_cloudagent/messaging/models/base_record.py", line 250, in retrieve_by_tag_filter
    cls.__name__, tag_filter, f", {post_filter}" if post_filter else ""
aries_cloudagent.storage.error.StorageNotFoundError: ConnRecord record not found for {'invitation_key': 'EK9krLvVxnb45y6qBTYHrLj2vpGJaruwJFbnG39PqURB'}, {'state': 'invitation', 'their_role': 'invitee'}
```

When checking the connection states for agent 1's  sub wallet 1, I can only see its connection info in agent 2's sub wallet 1, nothing for agent2's sub wallet 2 . However, when I check the connection states for agent 2's sub wallet 2, I can see connection state information even though the state shows "request". 

Below diagram shows what i've done:

**Agent 1 - sub wallet 1** (can see connection info for Agent 2's sub wallet 1)      -----------one-time connection established successfully----     **Agent 2 - sub wallet 1** 
**Agent 1 - sub wallet 1** (no info for Agent2's sub wallet 2)      -----------one-time connection establishment failed--------     **Agent 2 - sub wallet 2** (can see connection info for Agent1's sub wallet1)
**Agent 1 - sub wallet 2** (not used in my scenario)                    

Basically the connection states information in agent1's sub wallet 1 and agent2's sub wallet 2 are not showing the same thing. Behind the scene, I'm not sure if the same connection invite can be consumed more than once in ACA-Py. 

**Expected outcome**: if the same connection invite can only be consumed once, the state information in the two agent's information should be the same, so it won't confuse users.

**Comment by esune** [comment]
Did you turn on the `multi-use` flag when generating the OOB connection request? If not, the first agent receiving the invitation will be the only agent who can use it.

**Comment by yunxi-zhang** [comment]
Hello @esune, what is this "multi-use" flag? Is this a flag used when running an ACA-Py agent or when calling the API: `connections/create-invitation `? Any docs I can find for details?

**Comment by esune** [comment]
If you want to use the same invitation on more than one agent (if I understood correctly what you're doing) you need to add `multi_use=true` as query parameter when generating the connection invitation.
![image](https://user-images.githubusercontent.com/2395873/117471272-89b8f380-af0c-11eb-80d5-af747e4e1c21.png)

**Comment by yunxi-zhang** [comment]
Thanks for the info @esune.

Let's say we have agent 1 as invitation sender, agent 2 and agent 3 as invitation receiver. To establish a connection for below
1. Agent 1 - Agent 2
2. Agent 1 - Agent 3

I can either create an connection invitation in Agent 1 and reuse it for agents 2 and 3, or create 2 separate connection invitations, one for agent 2 and the other for agent 3. 

In theory and in practice, are there any differences between the two approaches?

**Comment by swcurran** [comment]
It depends whether you want to create a new invitation for establishing each connection.  For example, if you create a QR code for the invitation, and you post it on twitter, then you want to use a multi-use invitation, so many people can use it.  If you have a web-session with a single user, or perhaps you email the invitation to each individual contact, then you probably want to use a single-use invitiation that becomes invalid after its first use. 

Note that even if you use a single-use invitation, you should remember that it is a plain-text message that others might intercept, so you might want to be absolutely sure who you are talking to AFTER you establish the connection.

Closing this issue, but feel free to reopen if there is more to this.

**Comment by yunxi-zhang** [comment]
Thanks @swcurran. 

So, in terms of the connection establishment effect (i.e. agents can communicate, once the invitation from the sender agent has been successfully consumed by the receiver agent), there is no differences between the single-use and multi-use invitation, isn't it? 

You mentioned single-use invitation is a plain-text message, does it mean multi-use invitations are not plain-text messages?

**Comment by swcurran** [comment]
Both single- and multi-use invitations are plaintext messages.
